compiler/semtypes.nim

```diff
@@ -241,6 +241,12 @@ proc fitDefaultNode(c: PContext, n: PNode): PType =
   if n[^2].kind != nkEmpty:
     if expectedType != nil and oldType != expectedType:
       n[^1] = fitNodeConsiderViewType(c, expectedType, n[^1], n[^1].info)
+      annotateType(n[^1], expectedType, c.config) # infer types for default fields value
```

Comment:
We don't do this anywhere else in the compiler.

Reply:
Well, `changleType` works as well. Could we use that here? Because `semConstExpr` didn't infer the type from expectedType.

Reply:
`changeType` would be better, yes.

Reply:
Thanks! On it